bazelfe-core/src/bep_junit/bep_junit_app.rs

```diff
@@ -99,35 +107,60 @@ async fn main() -> Result<(), Box<dyn Error>> {
                 let output_folder = opt.junit_output_path.join(label_to_junit_relative_path(
                     r.test_summary_event.label.as_str(),
                 ));
-                std::fs::create_dir_all(&output_folder).expect("Make dir failed");
-
-                let files: Vec<&str> = r
-                    .test_summary_event
-                    .output_files
-                    .iter()
-                    .flat_map(|e| match e {
-                        bazelfe_protos::build_event_stream::file::File::Uri(uri) => {
-                            let p = uri
-                                .strip_prefix("file://")
-                                .unwrap_or_else(|| panic!("Wasn't a local file for {}", uri));
-                            if p.ends_with("/test.xml") {
-                                Some(p)
-                            } else {
-                                None
+
+                match std::fs::create_dir_all(&output_folder) {
+                    Ok(_) => {
+                        let files: Vec<&str> = r
+                            .test_summary_event
+                            .output_files
+                            .iter()
+                            .flat_map(|e| match e {
+                                bazelfe_protos::build_event_stream::file::File::Uri(uri) => {
+                                    let p = uri.strip_prefix("file://").unwrap_or_else(|| {
+                                        panic!("Wasn't a local file for {}", uri)
+                                    });
+                                    if p.ends_with("/test.xml") {
+                                        Some(p)
+                                    } else {
+                                        None
+                                    }
+                                }
+                                bazelfe_protos::build_event_stream::file::File::Contents(_) => None,
+                            })
+                            .collect();
+
+                        for (idx, f) in files.into_iter().enumerate() {
+                            match std::fs::metadata(f) {
+                          Ok(m) => {
+                            if m.size() > 0 {
+                              let output_file = output_folder.join(format!("test.{}.xml", idx));
+                              match std::fs::copy(f, output_file) {
+                                Ok(_) => (),
+                                Err(e) =>
+                                  println!("could not access metadata for test result {} at file {}.\nError {}",
+                                    r.test_summary_event.label,
+                                    f,
+                                    e)
+                              }
                             }
+                          }
+                          Err(e) =>
+                            println!("could not access metadata for test result {} at file {}.\nError {}",
```

Comment:
Does anything in here bubble back up? Should it?

Reply:
So this error means that when you went to access the size of the file, the filesystem operation failed.

It's for an intermediate test result that is reported by the Bazel event protocol log to exist.

We could halt the program here. I doubt this will happen. The cases we have seen look more like we can't write to the disk (maybe it's full, maybe we have some permission issue). Although if it is a permission issue, maybe we can't read.

I would say leave it as is now and we'll see how these spurious errors chance.

What do you think?

Reply:
yeah I think that's fine for now